### PR TITLE
Load tasks to pdf

### PR DESCRIPTION
**Pull request by BohdanBusko**
**STATUS**
READY

**DESCRIPTION**
screenshot from 2018-11-18 12-05-01

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs

### CODE REVIEW

File: app/controllers/account/tasks_controller.rb [hunk 1]
```diff
@@ -118,6 +118,15 @@ def uncomplete
     respond_to :js
   end
 
+  def load_to_pdf
```
Comment (AndriySolonyna) [inline app/controllers/account/tasks_controller.rb]: bad naming

File: app/pdfs/task_pdf_loader.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class TaskPdfLoader < Prawn::Document
+  def initialize(title, data)
+    super()
+    text title, align: :center, size: 24
+    move_down 15
+    text "Tasks: #{Date.today.beginning_of_week.strftime('%Y:%m:%d')} - #{Date.today.strftime('%Y:%m:%d')}", size: 16
```
Comment (AndriySolonyna) [inline app/pdfs/task_pdf_loader.rb]: use locales for custom date formats

File: spec/models/task_spec.rb [hunk 3]
```diff
@@ -57,4 +57,13 @@
       expect(Task.search_tasks(user.id, 'iajshdkas').count).to eq 0
     end
   end
+
+  describe '#this_week' do
```
Comment (AndriySolonyna) [inline spec/models/task_spec.rb]: # should be used only for instance methods, use . for scopes and class methods

File: app/controllers/account/tasks_controller.rb [hunk 4]
```diff
@@ -118,6 +118,15 @@ def uncomplete
...
+    pdf = TaskPdfLoader.new(parent.name, collection.this_week)
+
+    send_data pdf.render,
+      filename: "weekly_report_for_#{parent.name}.pdf",
+      type: 'application/pdf',
+      disposition: 'inline'
```
Comment (igorkasyanchuk) [inline app/controllers/account/tasks_controller.rb]: attachment so it will be downloaded automatically

File: app/pdfs/task_pdf_loader.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+class TaskPdfLoader < Prawn::Document
+  def initialize(title, data)
+    super()
```
Comment (igorkasyanchuk) [inline app/pdfs/task_pdf_loader.rb]: why ()

File: app/pdfs/task_pdf_loader.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+class TaskPdfLoader < Prawn::Document
```
Comment (igorkasyanchuk) [inline app/pdfs/task_pdf_loader.rb]: class is printing tasks of Project. Why it has name "Task..."
rename to something like ProjectTasksPdfReport

File: config/routes.rb [hunk 7]
```diff
@@ -33,6 +33,8 @@
           delete :unassign
           delete :remove_attachment
         end
+
+        get :load_to_pdf, on: :collection, defaults: { format: :pdf }
```
Comment (igorkasyanchuk) [inline config/routes.rb]: rename to report

File: title [hunk 8]
```diff
@@ -0,0 +1,73 @@
+%PDF-1.3
```
Comment (igorkasyanchuk) [inline title]: what is this?

File: app/pdfs/task_pdf_loader.rb (hunk 5 above)
Comment (BohdanBusko) [reply app/pdfs/task_pdf_loader.rb]: screenshot from 2018-11-19 09-29-16

File: app/assets/javascripts/search.js [hunk 9]
```diff
@@ -115,6 +116,27 @@ function globalSearch() {
   }
 }
 
+function onFocusChange() {
+  var search = document.getElementById('search');
+  var searchResults = document.getElementById('search-results');
+  var searchResultsMouse = false;
+
+  searchResults.addEventListener('mouseenter', () => { searchResultsMouse = true; console.log(searchResultsMouse); });
```
Comment (igorkasyanchuk) [inline app/assets/javascripts/search.js]: remove console.logs

**Review by igorkasyanchuk — APPROVED** [review]

**Pull request merged**